src/jit/morph.cpp

```diff
@@ -3813,6 +3813,21 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
             assert(size == 1);
 #endif
 #endif
+            // If 'expectRetBuffArg' is true then the next argument is the RetBufArg
+            // and we may need to change nextRegNum to the theFixedRetBuffReg
+            if (expectRetBuffArg)
+            {
+                assert(passUsingFloatRegs == false);
+              
+                if (hasFixedRetBuffReg())
+                {
+                    // Change the register used to pass the next argument to the fixed return buffer register
+                    nextRegNum = theFixedRetBuffReg();
+                }
+
+                // We no longer are expect the RetBufArg
```

Comment:
Looks like during the first iteration through args, we set nextRegNum to  x8 and don't increment integer arg reg num and to achieve it we are setting expectRetBufArg  to false.

A comment explaining this would be useful here as this is a complex routine.